jobserver/models/auditable_event.py

```diff
@@ -0,0 +1,53 @@
+from django.db import models
+from django.utils import timezone
+
+
+class AuditableEvent(models.Model):
+    """
+    A table to track changes to our models
+
+    We avoid ForeignKeys here so our audit log never interferes with our "live"
+    data.
+    """
+
+    class Type(models.TextChoices):
+        PROJECT_MEMBER_ADDED = "project_member_added", "User added"
+        PROJECT_MEMBER_REMOVED = "project_member_removed", "User removed"
+        PROJECT_MEMBER_UPDATED_ROLES = (
+            "project_member_updated_roles",
+            "User's roles updated",
+        )
+
+    old = models.TextField()
+    new = models.TextField()
+
+    # what is this record auditing?
+    # we use this field to make it explicit what's being audited and to look up
+    # related code, eg presenters
+    type = models.TextField(choices=Type.choices)  # noqa: A003
+
+    target_model = models.TextField()
+    target_field = models.TextField()
+    target_id = models.TextField()
```

Comment:
This is a question for future reference, rather than a suggested change.. why are the target_id and parent_id text fields rather than integers? As far as I can see, we're only using integers for the model ids.

Reply:
We have a handful of primary keys using ULIDs which are stored as strings, so this is just to cover all uses.